src/Bicep.LangServer/Settings/SettingsProvider.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.LanguageServer.Settings
+{
+    /// <summary>
+    /// Provider used to cache settings information received from <see cref="ConfigurationSettingsHandler"/>
+    /// </summary>
+    public class SettingsProvider : ISettingsProvider
+    {
+        private readonly Dictionary<string, bool> SettingsCache = new();
```

Comment:
>Dictionary [](http://example.com/codeflow?start=25&length=10)

Should this be a `ConcurrentDictionary`?

Reply:
Updated here:
https://github.com/Azure/bicep/commit/d7b1ad259e736f65aca7265b9737e4a203cb542b